contracts/connection-router/src/types.rs

```diff
@@ -81,63 +81,38 @@ impl KeyDeserialize for MessageID {
     }
 }
 
-/// cosmwasm cannot serialize tuples, so we need to convert [GlobalMessageId] into a struct
-#[cw_serde]
-struct GlobalMessageIdSerde {
-    chain_name: ChainName,
-    message_id: MessageID,
-}
-
+/// cosmwasm cannot serialize tuples, so we need to convert [CrossChainUid] into a struct
 #[cw_serde]
-#[serde(from = "GlobalMessageIdSerde", into = "GlobalMessageIdSerde")]
-pub struct GlobalMessageId(ChainName, MessageID);
-
-impl From<GlobalMessageId> for GlobalMessageIdSerde {
-    fn from(other: GlobalMessageId) -> Self {
-        Self {
-            chain_name: other.0,
-            message_id: other.1,
-        }
-    }
-}
-
-impl From<GlobalMessageIdSerde> for GlobalMessageId {
-    fn from(other: GlobalMessageIdSerde) -> Self {
-        Self(other.chain_name, other.message_id)
-    }
-}
-
-impl GlobalMessageId {
-    pub fn new(chain_name: ChainName, message_id: MessageID) -> Self {
-        Self(chain_name, message_id)
-    }
+pub struct CrossChainUid {
+    pub chain: ChainName,
+    pub id: MessageID,
```

Comment:
I still find it weird that the `Message` type has a `uid` field, within which there's a field called `MessageID`.